src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -202,7 +205,13 @@ private void ConfigureApplication(WebHostBuilderContext context, IApplicationBui
                 if (targetRouteBuilder is null)
                 {
                     // The app defined endpoints without calling UseRouting() explicitly, so call UseRouting() implicitly.
-                    app.UseRouting();
+                    app.UseRouting(overrideEndpointRouteBuilder: false);
+
+                    // Copy the endpoint route builder to the built application
+                    foreach (var item in app.Properties)
+                    {
+                        _builtApplication.Properties[item.Key] = item.Value;
+                    }
 
                     // An implicitly created IEndpointRouteBuilder was addeded to app.Properties by the UseRouting() call above.
                     targetRouteBuilder = GetEndpointRouteBuilder(app)!;
```

Comment:
This feels a little less hacky. This works, right?

```suggestion
                    // An implicitly created IEndpointRouteBuilder was addeded to app.Properties by the UseRouting() call above.
                    targetRouteBuilder = GetEndpointRouteBuilder(app)!;
                    // Copy the endpoint route builder to the built application
                    _builtApplication.Properties[EndpointRouteBuilderKey] = targetRouteBuilder;
```

Reply:
I think it'd work but I'm curious why that's better? Adding specific item to the `Property` bag still seems "hacky". I'm curious if we should keep the property bags on `_builtApplication` and `app` in sync? Would that be crossing too many wires?

Reply:
I'd rather do only what's necessary rather than do extra stuff. If there was a scenario this fixed, then fine.